subsys/bluetooth/mesh/rpl_storage.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <string.h>
+#include <stdbool.h>
+
+#include <bluetooth/mesh.h>
+
+#include "net.h"
+#include "rpl.h"
+
+/* Covers no persistent storage usecase */
+static struct bt_mesh_rpl replay_list[CONFIG_BT_MESH_CRPL];
```

Comment:
Just merge `rpl_storage.c` and `rpl_strg_settings.c` 😃

Reply:
But we have to keep the "zephyr_library_sources_ifdef..." so we can remove the file if none of the choices are made.

Reply:
Ok, I think I get it. You want me to keep `rpl_strg_settings`as default in both the `Settings` and `No persistent storage` usecase? And let `CONFIG_BT_SETTINGS` decide if there is persistent stoarage available or not? If this assumption is correct, I dont think it will tamper with the main idea @Balaklaka. It might make the Kconfig choise a little more convoluted IMO, since the `No persistent` choise will be nested within the Settings choise and depend on if `CONFIG_BT_SETTINGS` is set or not. On the other hand my original choise bloats the code a bit so im not opposed to doing it the way like pavel suggested.

Reply:
I guess something like with BT_LL_CHOICE should work here as well:

- in zephyr:
https://github.com/zephyrproject-rtos/zephyr/blob/e9d3791e3f33b7b95f57de80a85c374660680a1c/subsys/bluetooth/controller/Kconfig#L90-L105
- in sdk-nrf:
https://github.com/nrfconnect/sdk-nrf/blob/b190134d8b96cb94c4d20d5a42375a8927134a45/subsys/bluetooth/controller/Kconfig#L7-L35

We may even try to put emds version of rpl_storage.c in sdk-nrf repo. At least I would recommend to try this now, before proposing the final change to upstream.

Reply:
Yes, I think that was the main idea proposed by Trond. So you want me to test if can link in a rpl_strg_mock.c from sdk-nrf to this altered rpl implementation then to see if it works?

Reply:
Yes, I think we can prototype this now and even check if it works with emds. I guess you can just create a `rpl_storage_emds.c` or so.

Reply:
I have tested this by cherrypicking this PR on top of man-rev, added a mock of rpl_storage_emds.c in sdk-nrf subsystem and added the necessary kconfig/cmakelist options. As far as I can see it works fine when testing it on light sample 👍